### PR TITLE
SC-9544: Journald bugfix + Added lowercase filter

### PR DESCRIPTION
**Pull request by adnanrahic**
This PR adds:

- [x] lowercase filter
- [x] updated Journald setup script

### CODE REVIEW

File: config/examples/output-filter-lowercase-fields.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    module: lowercase-fields
+    # JS regular expression to match log source name
+    matchSource: !!js/regexp .*
+    allFields: true # this will lowercase all fields
+    fields:
+      - fieldName: SELINUX_CONTEXT # this will lowercase only certain fields, this is an array and you can specify multiple fields
+
+# Exmple, input: 
+# {"user": "{ name: root }", "message": "Client connect: root", "originalLine": "INFO Client connect: root"}
+# Example, output
```
Comment (radu-gheorghe) [inline config/examples/output-filter-lowercase-fields.yml:12]: Are you sure this is what happens here? renaming `user` to `user-object`?

File: lib/plugins/output-filter/journald-format.js [hunk 2]
```diff
@@ -134,7 +134,7 @@ function applySematextCommonSchema (
             Object.assign(log[type], data)
           }
           callback(null, log)
-        }.bind(this)
+        }
```
Comment (radu-gheorghe) [inline lib/plugins/output-filter/journald-format.js:137]: I have no idea what the bind() did here. Is it easy to explain? :)

File: lib/plugins/output-filter/lowercase-fields.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    config.allFields = false
+  }
+
+  try {
+    if (!config.matchSource.test(context.sourceName || data.logSource)) {
+      return callback(null, data)
+    }
+
+    if (config.allFields === true) {
+      const fields = Object.keys(data)
```
Comment (radu-gheorghe) [inline lib/plugins/output-filter/lowercase-fields.js:19]: I take it that this only works for root keys? Does it work for something like `myobject.WHY_ARE_YOU_YELLING_THE_FIELD_NAME` ?

**Review by radu-gheorghe — APPROVED** [review]
Left a few minor comments, but otherwise LGTM.

File: config/examples/output-filter-lowercase-fields.yml (hunk 1 above)
Comment (adnanrahic) [reply config/examples/output-filter-lowercase-fields.yml:12]: Ah damn, leftover comments. Will change.

File: lib/plugins/output-filter/journald-format.js (hunk 2 above)
Comment (adnanrahic) [reply lib/plugins/output-filter/journald-format.js:137]: Sure. The `bind()` method is a prototype method on the function prototype chain in JavaScript, that returns a new function with a bound scope. It binds the `this` context into the bound function. There is no outer scope to the `this` which is referred here, hence making it redundant.

From [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind):
> The `bind()` method creates a new function that, when called, has its this keyword set to the provided value, with a given sequence of arguments preceding any provided when the new function is called.

File: lib/plugins/output-filter/lowercase-fields.js (hunk 3 above)
Comment (adnanrahic) [reply lib/plugins/output-filter/lowercase-fields.js:19]: Yes, only root keys for now. Let me think about if I can edit it to cascade the lowercasing across all keys as well.

**Comment by adnanrahic** [comment]
Conclusion: I'll add the cascading lowercasing some other time. It's not relevant right now for solving this issue. I've updated the descriptions to make sure we tell users it will only lowercase root fields.

**Pull request merged**